enhancements/installer/byo-custom-dns.md

```diff
@@ -0,0 +1,537 @@
+same name and annotations as one created by the cluster-ingress-operator, ARO
+is able to create an Ingress LB during the bootstrap phase. The non-cloud DNS
+could be pre-configured to resolve *.apps service to this pre-defined static
+IP.
+
+This would work well except for the following 2 reasons:
+1. AWS does not refer to the `LoadBalancerIP` input value before it creates an
+Ingress LB. So, the above solution only would work for Azure and GCP and not
+for AWS.
+
+2. Based on [K8s API documentation] (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#servicespec-v1-core), the `LoadBalancerIP` field would soon be deprecated.
+
+### API Extensions
+
+1. Install config is augmented to allow the customer to specify if a custom
+DNS solution is being used. In addition, the customer would be able to indicate
+the LB type to be used for API and API-Int traffic.
+
+```yaml
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.9.2
+  creationTimestamp: null
+  name: installconfigs.install.openshift.io
+spec:
+  group: install.openshift.io
+  names:
+    kind: InstallConfig
+    listKind: InstallConfigList
+    plural: installconfigs
+    singular: installconfig
+  scope: Namespaced
+  versions:
+  - name: v1
+    schema:
+<snip>
+                    aws:
+                      description: AWS is the configuration used when installing on
+                        AWS.
+                      properties:
+                        userConfiguredDNSLB:
+                          description: UserProvidedDNSLB allows the user to specify the LB
+                            Names for API and API-Int LBs. The user's custom DNS solution
+                            is expected to be have been pre-configured with these LBs.
+                          properties:
+                            userConfiguredDNS:
+                              description: UserConfiguredDNS is set to `Enabled` when the customer
+                                wants to use a DNS solution external to the cluster and OpenShift is
+                                not allowed to configure it. The default value is `Disabled`.
+                              enum:
+                              - ""
+                              - Enabled
+                              - Disabled
+                              type: string
+                            apiLBName:
+                              description: apiLBName is set to the LB Name of the API LB
+                              type: string
+                            apiLBType:
+                              description: apiLBType is set to the LB Type of the API LB. If
+                                unspecified, it will default to the NLB.
+                              enum:
+                              - ""
+                              - NLB
+                              - ALB
```

Comment:
Do we support ALB in a common IPI install? If I remember right, we only support NLB, right?

Reply:
@sadasu any comments on this?

Reply:
Thanks for your comments.

If that is still the case, AFAIK, ALB and NLB has different configurations, if IPI install creates and configures an NLB, once user pre-create an ALB, I have concern IPI installer can configure it correctly.

Reply:
That is valid. Allowing the user to pre-create an ALB would involve updating the Installer to be able to configure the ALB. I added it in the hopes of making it future proof, in the event that we wanted to also add BYO LB. 
But, in retrospect, I do not want to do that now. I am updating the api to just take the LB Name of the NLB that is created by the user.